platform-emf/src/main/java/com/softicar/platform/emf/data/table/filter/string/EmfDataTableStringFilterNode.java

```diff
@@ -40,14 +40,29 @@ public IEmfDataTableFilter<R> createFilter() {
 				return new EqualsTextFilter<>(column.getDataColumn(), filterText, resetter);
 			case MATCHES_REGEXP:
 				return new MatchesRegexpFilter<>(column.getDataColumn(), filterText, resetter);
+			case EMPTY:
+				throw new UnsupportedOperationException();
+			case NOT_EMPTY:
+				throw new UnsupportedOperationException();
 			}
-
 			throw new SofticarUnknownEnumConstantException(filterType);
 		} else {
+			if (filterType.equals(EmfDataTableStringFilterType.EMPTY)) {
+				return new EmptyTextFilter<>(column.getDataColumn(), resetter);
+			} else if (filterType.equals(EmfDataTableStringFilterType.NOT_EMPTY)) {
+				return new NotEmptyTextFilter<>(column.getDataColumn(), resetter);
+			}
 			return new EmfDataTableNopFilter<>(resetter);
 		}
 	}
 
+	@Override
+	public void refresh() {
+
+		super.refresh();
+		filterInput.refresh(filterTypeSelect.getSelectedValue());
```

Comment:
not 100% sure if this is the cleanest solution, but I have no other suggestion, just saying

Reply:
definitely not the cleanest, but the best without bigger refactorings imho